packages/ui/components/input-amount/src/parsers.js

```diff
@@ -30,10 +31,15 @@ function round(value, decimals) {
  * @param {FormatOptions} [givenOptions] Locale Options
  */
 export function parseAmount(value, givenOptions) {
+  const unmatchedInput = value.match(/[^0-9,.\- ]/g);
+  if (unmatchedInput && givenOptions?.mode !== 'pasted') {
```

Comment:
Should we explain the pasted behavior and is there a test for it?

Reply:
yes there is a test for it.
Explanation won't hurt.